mpisppy/cylinders/xhatshufflelooper_bounder.py

```diff
@@ -83,7 +83,8 @@ def _vb(msg):
             return False
         _vb(f"    Feasible {snamedict}, obj: {obj}")
 
-        update = self.update_if_improving(obj)
+        # the xhatter updates the cache in the opt object for us
```

Comment:
I don't understand this comment

Reply:
It happens [here](https://github.com/Pyomo/mpi-sppy/blob/a9c522173ca37459d2b311de5caf2ed08c5b38ad/mpisppy/extensions/xhatbase.py#L205) and [here](https://github.com/Pyomo/mpi-sppy/blob/a9c522173ca37459d2b311de5caf2ed08c5b38ad/mpisppy/extensions/xhatbase.py#L232).

On critical thing is that `SPBase` now tracks the best solution found so far, as opposed to the incumbent spoke.

Reply:
Basically, you want to cache the solution *before* the xhatter restores the nonants.